### PR TITLE
Added async function chain

### PR DESCRIPTION
**Pull request by RobertAKARobin**
I included a "long form" as well. I think providing an explanation in this way may be very helpful to beginners, for whom this is a great resource. GitHub's `<details>` functionality is a little-known secret.

### CODE REVIEW

File: snippets/chain-async-functions.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+### Chain asynchronous functions
+
+Loop through an array of functions containing asynchronous events, calling `next` when each asynchronous event has completed.
+
+```js
+const chainAsync = fns => {
+	let curr = 0; const next = () => fns[curr++](next); next()
+}
```
Comment (Chalarangelo) [inline snippets/chain-async-functions.md]: Provide an example in a comment, below the declaration, please.

File: snippets/chain-async-functions.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+<details>
```
Comment (Chalarangelo) [inline snippets/chain-async-functions.md]: Indeed `<details>` is a pretty cool way to explain, but it doesn't play on any MS browser, so this is probably a bit of a problem. Maybe linking to an explanation or article would be preferrable.

File: snippets/chain-async-functions.md (hunk 1 above)
Comment (RobertAKARobin) [reply snippets/chain-async-functions.md]: @Chalarangelo Done.

File: snippets/chain-async-functions.md (hunk 2 above)
Comment (RobertAKARobin) [reply snippets/chain-async-functions.md]: I wasn't aware of the MS limitation. Shouldn't be surprised. Thanks.

**Pull request merged**